apple/internal/ios_rules.bzl

```
@@ -542,6 +547,75 @@ def _ios_app_clip_impl(ctx):
         link_result.binary_provider,
     ] + processor_result.providers
 
+def _ios_dylib_impl(ctx):
+    """Implementation of the ios_dylib rule."""
+    link_result = linking_support.register_linking_action(ctx)
+    binary_artifact = link_result.binary_provider.binary
+    debug_outputs_provider = link_result.debug_outputs_provider
+
+    actions = ctx.actions
+    bin_root_path = ctx.bin_dir.path
+    bundle_name, bundle_extension = bundling_support.bundle_full_name_from_rule_ctx(ctx)
+    executable_name = bundling_support.executable_name(ctx)
+    entitlements = entitlements_support.entitlements(
+        entitlements_attr = getattr(ctx.attr, "entitlements", None),
+        entitlements_file = getattr(ctx.file, "entitlements", None),
+    )
+    label = ctx.label
+    platform_prerequisites = platform_support.platform_prerequisites_from_rule_ctx(ctx)
+    predeclared_outputs = ctx.outputs
+    rule_descriptor = rule_support.rule_descriptor(ctx)
+    rule_executables = ctx.executable
+
+    debug_outputs_partial = partials.debug_symbols_partial(
+        actions = actions,
+        bin_root_path = bin_root_path,
+        bundle_extension = bundle_extension,
+        bundle_name = bundle_name,
+        debug_outputs_provider = debug_outputs_provider,
+        dsym_info_plist_template = ctx.file._dsym_info_plist_template,
+        executable_name = executable_name,
+        platform_prerequisites = platform_prerequisites,
+        rule_label = label,
+    )
+
+    processor_result = processor.process(
+        ctx = ctx,
+        actions = actions,
+        bundle_extension = bundle_extension,
+        bundle_name = bundle_name,
+        bundle_post_process_and_sign = False,
+        entitlements = entitlements,
+        executable_name = executable_name,
+        partials = [debug_outputs_partial],
+        platform_prerequisites = platform_prerequisites,
+        predeclared_outputs = predeclared_outputs,
+        provisioning_profile = getattr(ctx.file, "provisioning_profile", None),
+        rule_descriptor = rule_descriptor,
+        rule_executables = rule_executables,
+        rule_label = label,
+    )
+    output_file = ctx.actions.declare_file(ctx.label.name + ".dylib")
```

```suggestion
    output_file = actions.declare_file(ctx.label.name + ".dylib")
```

this is what macos_dylib does, seems like with this and the change below there can be 0 diff between them